Standalone setup navigation (#938)
agentgateway/website: assets/agw-docs/standalone/setup/update.md

> Change the configuration of a running agentgateway instance. The steps differ by installation method, because each method delivers the configuration file to the proxy in a different way.

Reviewer: ```suggestion
The steps differ by installation method, because each method delivers the configuration file to the proxy in a different way.
```

## About

In standalone mode, one configuration file is the source of truth for the proxy. Agentgateway watches that file and reloads it when the contents change, so most changes take effect without restarting the process.

```txt
INFO state_manager  Watching config file: /config/config.yaml
INFO state_manager  loaded config from File("/config/config.yaml")
```

Two things are worth knowing before you edit.

* **Not every field reloads.** The top-level `config` section holds startup settings, such as `adminAddr`, `storage`, `database`, `logging`, and `tracing`. Agentgateway applies those only when the process starts, with the exception of `config.modelCatalog`, which does reload. Everything else, including `gateways`, `routes`, `llm`, `mcp`, and `ui`, reloads in place. For more information, see [Fields that require a restart](#restart-required).

> * **The UI might write to this file.** In the default storage mode in binary and Docker deployments, agentgateway writes the resources that you manage in the admin UI back to the same file. Your file is an output as well as an input. To keep the file as the only writer, or to send UI edits to a database instead, see [Configuration storage]({{< link-hextra path="/setup/storage/" >}}).

Reviewer: ```suggestion
* **The UI might write to this file.** In the default storage mode in binary and Docker deployments, agentgateway writes the resources that you manage in the admin UI back to the same file. Your file is an output as well as an input. To keep the file read-only, or to send UI edits to a database instead, see [Configuration storage]({{< link-hextra path="/setup/storage/" >}}).
```

1. Edit the file in the directory or at the path that you mounted. If you mounted a directory at `/config`, the file is `config.yaml` inside it.

   ```sh
   vi agentgateway-config/config.yaml
   ```

2. Save the file. Agentgateway reloads it inside the container. Check the container logs to confirm.

   ```sh
   docker logs <container-name> | tail -5
   ```

   Example output:

   ```txt
   INFO state_manager  loaded config from File("/config/config.yaml")
   ```

3. **Optional**: If you mounted the configuration read-only, or if the change is to the `config` section, restart the container instead.

   ```sh
   docker restart <container-name>
   ```

4. Confirm that the running configuration includes your change. For more information, see [Verify an update](#verify).

## Helm {#helm}

With the Helm chart, you do not edit a file on the proxy. The `config` Helm value holds the entire agentgateway configuration file, and the chart renders it into the ConfigMap that the pod mounts. To change the configuration, change your values and upgrade the release.

> [!WARNING]

> > The default replicas for your agentgateway Deployment is `1`. To avoid a brief interruption in traffic during the rollout, increase the `replicaCount` setting To keep a pod serving traffic while the new pod starts, set `replicaCount` to a value greater than `1`.

Reviewer: ```suggestion
> The default replicas for your agentgateway Deployment is `1`. To avoid a brief interruption in traffic during the rollout, increase the `replicaCount` setting to a value greater than `1`. This way, one of the pods can continue serving traffic while the new configuration is rolled out.
```

         port: 4000
     routes:
     - matches:
       - path:
           pathPrefix: /
       backends:
       - host: httpbin.httpbin.svc.cluster.local:8000
     config:                  # agentgateway field: agentgateway's own 'config' section
       logging:
         level: info
   replicaCount: 2            # run multiple replicas of the agentgateway proxy
   EOF
   ```

2. Pass the file to Helm during the upgrade.

   {{< reuse "agw-docs/standalone/helm-upgrade-command.md" >}}

3. Confirm that the ConfigMap holds your change.

   ```sh
   kubectl get configmap {{< reuse "agw-docs/standalone/helm-standalone-release.md" >}}-config \
     -n {{< reuse "agw-docs/snippets/namespace.md" >}} -o jsonpath='{.data.config\.yaml}'
   ```

> [!IMPORTANT]
> The upgrade replaces the whole configuration file, not just the fields that you changed. A value that you leave out of the values file returns to its chart default, so pass your complete values file on every upgrade, or use `--reuse-values` to keep the values from the previous revision. In `database` storage mode, the upgrade replaces only the ConfigMap baseline, and the resources that the UI stored in the database are unaffected.

> ## Fields that require a restart {#restart-required}

Reviewer: thinking this section can be removed as you describe that in the beginning and also as part of each binary, docker, helm section

Restart agentgateway for the installation method that you use.

{{< tabs >}}
{{% tab name="Binary" %}}
1. Stop the current process, such as with `ctrl+c`.
2. Restart the process.
   ```sh
   agentgateway -f config.yaml
   ```
{{% /tab %}}
{{% tab name="Docker" %}}
```sh
docker restart <container-name>
```
{{% /tab %}}
{{% tab name="Helm" %}}
Choose from the following options:
- To restart with a configuration change, upgrade the Helm installation:
  {{< reuse "agw-docs/standalone/helm-upgrade-command.md" >}}
- To restart without a configuration change:
  ```sh
  kubectl rollout restart deploy/{{< reuse "agw-docs/standalone/helm-standalone-release.md" >}} -n {{< reuse "agw-docs/snippets/namespace.md" >}}
  ```
{{% /tab %}}
{{< /tabs >}}

> ## Verify an update {#verify}

Reviewer: this could be incorporated into the binary, docker, helm sections and removed from here. but I leave that up to you
